### PR TITLE
Movies in ISOs

### PR DESCRIPTION
**Pull request by TheCycoONE**
If you are playing with an ISO hosp directory you can now have in game movies!

### CODE REVIEW

File: CorsixTH/Src/iso_fs.cpp [hunk 1]
```diff
@@ -423,36 +423,35 @@ bool iso_filesystem::file_metadata_less(const file_metadata& lhs,
...
-  iso_directory_iterator dir_iter(pDirEnt, pDirEnt + iDirEntsSize);
-  iso_directory_iterator end_iter(pDirEnt + iDirEntsSize,
-                                  pDirEnt + iDirEntsSize);
+  iso_directory_iterator dir_iter(pDirEnt, pDirEnt + dirEntsSize);
+  iso_directory_iterator end_iter(pDirEnt + dirEntsSize, pDirEnt + dirEntsSize);
   for (; dir_iter != end_iter; ++dir_iter) {
     const iso_file_entry& ent = *dir_iter;
     if (ent.flags & def_directory) {
       // The names "\x00" and "\x01" are used for the current directory
       // the parent directory respectively. We only want to visit these
```
Comment (Alberth289346) [inline CorsixTH/Src/iso_fs.cpp:443]:
```suggestion
      // The names "\x00" and "\x01" are used for the current directory and
      // the parent directory respectively. We only want to visit these
```
Probably fits as new line 438 is longer.

File: CorsixTH/Src/th_lua_iso.cpp [hunk 2]
```diff
@@ -80,11 +80,27 @@ int l_isofs_file_size(lua_State* L) {
...
+  iso_filesystem::file_handle iFile = pSelf->find_file(sFilename);
+  if (!iso_filesystem::is_handle_good(iFile)) {
+    lua_pushnil(L);
+    lua_pushnil(L);
+    lua_pushfstring(L, "Could not find \'%s\' in .iso image", sFilename);
+    return 3;
+  }
+  uint32_t start = pSelf->get_file_start(iFile);
+  lua_pushinteger(L, start);
+  lua_pushinteger(L, start + pSelf->get_file_size(iFile));
```
Comment (Alberth289346) [inline CorsixTH/Src/th_lua_iso.cpp:95]: It isn't documented, but `start` looks like an inclusive lower bound. (If `.sector` is 0 you get 0 as return value).
Assuming "size" is a proper size, the second value is thus an exclusive upper bound.

In Lua it is normal to use inclusive upper bounds. (and start at 1, but I can see that might be a bad idea).

As it's used for building a string, it's probably fine just a bit strange from a Lua point of view.

**Comment by lewri** [comment]
I'll need to make an ISO copy of the game. Should I wait for you to go through Albert's comments before testing?

File: CorsixTH/Lua/filesystem.lua [hunk 3]
```diff
@@ -219,6 +220,15 @@ function FileSystem:fileSize(virtual_path, ...)
   return lfs.attributes(s, "size")
 end
 
+function FileSystem:fileUri(virtual_path, ...)
+  virtual_path = getFullPath(virtual_path, ...)
+  if self.provider then
+    local fileStart, fileEnd = self.provider:fileOffsets(virtual_path)
+    return('subfile,,start,' .. fileStart .. ',end,' .. fileEnd .. ',,:' .. self.physical_path)
```
[the remark this answers is not in the document]
Comment (TheCycoONE) [reply CorsixTH/Lua/filesystem.lua:227]: Yes we do, I'll fix

File: CorsixTH/Src/th_lua_iso.cpp (hunk 2 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua_iso.cpp:95]: I'd rather cover this one with documentation instead of converting back and forth.

**Comment by TheCycoONE** [comment]
> I'll need to make an ISO copy of the game. Should I wait for you to go through Albert's comments before testing?

I'm actually more concerned with regressions on the non-ISO version. I can cover the ISO if you prefer.

File: CorsixTH/Src/th_lua_iso.cpp (hunk 2 above)
Comment (TheCycoONE) [reply CorsixTH/Src/th_lua_iso.cpp:95]: @Alberth289346 for what it's worth, lua doesn't extend this convention to file offsets. https://www.lua.org/manual/5.4/manual.html#pdf-file:seek

**Comment by TheCycoONE** [comment]
Updated with suggestions and some additional documentation. Tested in Windows.

File: CorsixTH/Src/th_lua_iso.cpp (hunk 2 above)
Comment (Alberth289346) [reply CorsixTH/Src/th_lua_iso.cpp:105]: Interesting :)

**Review by Alberth289346 — APPROVED** [review]
LGTM

**Review by lewri — APPROVED** [review]
Non-ISO movies still all working

**Pull request merged**